### PR TITLE
Add more numpy-style attributes to HDF5Matrix

### PR DESCRIPTION
**Pull request by moi90**
When replacing a numpy array with HDF5Matrix, certain attributes are required. I added `dtype`, `ndim` and `size` and updated these including `shape` with the corresponding doc strings from h5py.Dataset.

### CODE REVIEW

File: keras/utils/io_utils.py [hunk 1]
```diff
@@ -96,7 +96,24 @@ def __getitem__(self, key):
...
+    
+    @property
+    def dtype(self):
+        """Numpy dtype representing the datatype"""
+        return self.data.dtype
+    
+    @property
+    def ndim(self):
+        """Numpy-style attribute giving the number of dimensions"""
+        self.data.size
```
Comment (Danielhiversen) [inline keras/utils/io_utils.py]: Remove this line?

**Comment by fchollet** [comment]
It's a reasonable change, but you should add unit tests for it (for coverage, and to avoid breaking it in the future), and write docstrings that are aligned with docstrings everywhere else.

**Comment by moi90** [comment]
`@property`s are sparsely documented in the code base, so it's not easy for me to infer a documentation style. Given that these are setters, a `Returns´ section would be redundant, because it would say the same as the first line in the docstring, wouldn't it? Is there a policy on this? Maybe it should be included in the [CONTRIBUTING.md](https://github.com/fchollet/keras/blob/master/CONTRIBUTING.md).

**Comment by fchollet** [comment]
Yes, they should have a `Returns` section. The one-line description should end with a period.

**Comment by moi90** [comment]
I updated the pull request accordingly. One more question: Are docstrings supposed to be first person ("Gets a numpy-style shape tuple") or third-person ("Get a numpy-style shape tuple")?

**Comment by fchollet** [comment]
Third-person ("**Gets** a...")

Additionally, please close the docstring (`"""`) on a new line.

**Comment by fchollet** [comment]
LGTM

**Comment by moi90** [comment]
Thanks for accepting my first contribution to Keras!

**Pull request merged**